firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/IdManager.java

```diff
@@ -171,19 +171,29 @@ private String readCachedCrashlyticsInstallId(SharedPreferences prefs) {
     return prefs.getString(PREFKEY_INSTALLATION_UUID, null);
   }
 
-  /** Makes a blocking call to query FID. If the call fails, logs a warning and returns null. */
-  @Nullable
-  @VisibleForTesting(otherwise = VisibleForTesting.PACKAGE_PRIVATE)
-  public String fetchTrueFid() {
-    Task<String> currentFidTask = firebaseInstallationsApi.getId();
-    String currentFid = null;
+  /**
+   * Makes a blocking call to query the Firebase installation id and Firebase authentication token.
+   *
+   * <p>If either call fails for any reason, logs a warning and sets a null value for that field.
+   */
+  @NonNull
+  public FirebaseInstallationId fetchTrueFid() {
+    String fid = null;
+    String authToken = null;
 
+    // Fetch the auth token first, so the fid will be validated.
+    try {
+      authToken = awaitEvenIfOnMainThread(firebaseInstallations.getToken(false)).getToken();
```

Comment:
Any reason for Crashlytics we need to do this on main thread?

Reply:
There is a flow when the SDK failed to initialize async on the last launch, it will attempt to initialize sync on the next launch.